com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -1118,6 +1172,18 @@ public bool WorldPositionStays
                 set => ByteUtility.SetBit(ref m_BitField, 5, value);
             }
 
+            /// <summary>
+            /// Even though the server sends notifications for NetworkObjects that get
+            /// destroyed when a scene is unloaded, we want to synchronize this so
+            /// the client side can use it as part of a filter for automatically migrating
+            /// to the current active scene when its scene is unloaded. (only for dynamically spawned)
+            /// </summary>
+            public bool DestroyWithScene
+            {
+                get => ByteUtility.GetBit(m_BitField, 6);
```

Comment:
The 'magic' bitField bit indexes numbers should really be consts or defines somewhere, I can see its how its done above though 😀 .  Might be worth another PR after this one to just tidy that up or pop a task somewhere. Not needed for this PR though.

Reply:
Yeah, there is history behind the entire `SceneObject` structure where it originally was to handle the initial serialization and synchronization of in-scene placed `NetworkObject`s.  In all reality, this area of code has been slowly improving over time (much better than when it was first implemented).
But we definitely should have another pass (like we are doing with the systems) at classes (i.e. NetworkObject, NetworkBehaviour, etc) where we make adjustments that don't impact users but can help to shave off processing cycles.
_For the most part, the ByteUtility does aggressive inlining so the bit shifting ends up not being that big of a blip on the over-all processing cost radar relative to everything else._

Reply:
I was thinking more for readability so that line would read

Reply:
Yeah... 😸 (_totally missed that point_)
I totally agree... that is a good idea. 👍